test/builders/statediffs.go

```diff
@@ -1,28 +1,40 @@
 package builders
 
 import (
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"github.com/orbs-network/orbs-spec/types/go/protocol"
 )
 
 type sdiff struct {
 	builder *protocol.ContractStateDiffBuilder
+	records []*protocol.StateRecordBuilder
 }
 
 func ContractStateDiff() *sdiff {
 	return &sdiff{
 		builder: &protocol.ContractStateDiffBuilder{
 			ContractName: "BenchmarkToken",
-			StateDiffs: []*protocol.StateRecordBuilder{
-				{Key: []byte("amount"), Value: []byte{10}},
-			},
 		},
 	}
 }
 
 func (s *sdiff) Build() *protocol.ContractStateDiff {
+	if len(s.records) == 0 {
+		s.records = append(s.records, &protocol.StateRecordBuilder{Key: []byte("amount"), Value: []byte{10}})
```

Comment:
Why did you move the amount to the build? (Same as the comment above.)

Reply:
because we need to be able to generate a valid state diff. This is only called if dev did not call `WithStringRecord()`

Reply:
an alternate way would be to create the builder with the default record, and on the first WithStringRecord() to remove that default value before appending the new pair..
@electricmonk  what do you think?

Reply:
how do you know it's a default value? you'll need to have a flag that indicates that the value is a default. I'd prefer to leave it as it is.